Testall fails in ambiguous

On

```
Julia Version 0.5.0-dev+4047
Commit 8f4e33f* (2016-05-12 08:42 UTC)
Platform Info:
  System: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, haswell)
```


testall fails with (showing end of output only, rest available)
```

	From worker 2:	     * checked               in   8.12 seconds, maxrss 1046.56 MB
	From worker 2:	     * inline                in   0.18 seconds, maxrss 1046.68 MB
	From worker 2:	     * boundscheck           in   0.06 seconds, maxrss 1046.77 MB
	From worker 3:	     * misc                  in  12.97 seconds, maxrss 1117.50 MB
WARNING: Method definition ambig(Any, Integer) in module Main at /home/colin/Downloads/julia/test/ambiguous.jl:4 overwritten at /home/colin/Downloads/julia/test/ambiguous.jl:58.
	From worker 3:	     * ambiguous            Skipping Base.<|
	From worker 3:	Skipping Base.>:
	From worker 3:	Skipping Base.active_repl
	From worker 3:	Test Failed
	From worker 2:	     * error                 in   1.83 seconds, maxrss 1047.00 MB
	From worker 3:	  Expression: isempty(detect_ambiguities(Core,Base; imported=true))
	From worker 6:	     * dsp                   in  11.69 seconds, maxrss  230.16 MB
	From worker 2:	     * fft                   in  18.92 seconds, maxrss 1083.67 MB
	From worker 6:	     * examples              in  13.56 seconds, maxrss  275.24 MB
	From worker 5:	     * parallel              in  41.54 seconds, maxrss 1284.36 MB
	From worker 4:	     * cmdlineargs           in  99.29 seconds, maxrss 1039.40 MB
Exception running test ambiguous :
On worker 3:
LoadError: There was an error during testing
 in record at ./test.jl:321
 in do_test at ./test.jl:219
 [inlined code] from ./int.jl:33
 in include_string at ./loading.jl:380
 in include_from_node1 at ./loading.jl:429
 [inlined code] from ./util.jl:179
 in runtests at /home/colin/Downloads/julia/test/testdefs.jl:7
 in #16 at /home/colin/Downloads/julia/test/runtests.jl:36
 [inlined code] from ./promotion.jl:248
 in #308 at ./multi.jl:1030
 in run_work_thunk at ./multi.jl:754
 [inlined code] from ./multi.jl:1030
 in #307 at ./event.jl:46
while loading /home/colin/Downloads/julia/test/ambiguous.jl, in expression starting on line 110
ERROR: LoadError: Some tests exited with errors.
 in (::##11#19)() at /home/colin/Downloads/julia/test/runtests.jl:64
 in cd(::##11#19, ::String) at ./file.jl:48
 in include_from_node1(::String) at ./loading.jl:426
 in process_options(::Base.JLOptions) at ./client.jl:263
 in _start() at ./client.jl:319
while loading /home/colin/Downloads/julia/test/runtests.jl, in expression starting on line 13
Makefile:12: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:568: recipe for target 'testall' failed
make: *** [testall] Error 2
```

--- a/base/multi.jl
+++ b/base/multi.jl
@@ -1212,7 +1212,7 @@ function init_worker(cookie::AbstractString, manager::ClusterManager=DefaultClus
     assert(isempty(PGRP.refs))
     assert(isempty(client_refs))
 
-    # System is started in head node mode, cleanup entries related to the same
+    # System is started in head node mode, cleanup related entries
     empty!(PGRP.workers)
     empty!(map_pid_wrkr)
 

--- a/deps/Makefile
+++ b/deps/Makefile
@@ -310,7 +310,7 @@ $5/$$($2_SRC_DIR)/$3: $$($2_SRC_FILE)
 endif # DEPS_GIT
 
 distclean-$1:
-	-rm -rf $5/$$($2_SRC_DIR) $$($2_SRC_FILE) $$(BUILDDIR)/$1
+	-rm -rf $5/$$($2_SRC_DIR) $$($2_SRC_FILE) $$(BUILDDIR)/$1 $$(BUILDDIR)/$1-$$($2_SHA1)
 endef
 
 

--- a/test/ambiguous.jl
+++ b/test/ambiguous.jl
@@ -107,7 +107,7 @@ ambs = detect_ambiguities(Ambig5)
 @test length(ambs) == 2
 
 # Test that Core and Base are free of ambiguities
-@test isempty(detect_ambiguities(Core, Base; imported=true))
+@test (x->(isempty(x) || println(x)))(detect_ambiguities(Core, Base; imported=true))
 
 amb_1(::Int8, ::Int) = 1
 amb_1(::Integer, x) = 2

--- a/test/core.jl
+++ b/test/core.jl
@@ -1773,6 +1773,7 @@ type A6142 <: AbstractMatrix{Float64}; end
 +{TJ}(x::A6142, y::UniformScaling{TJ}) = "UniformScaling method called"
 +(x::A6142, y::AbstractArray) = "AbstractArray method called"
 @test A6142() + I == "UniformScaling method called"
++(x::A6142, y::Range) = "Range method called" #16324 ambiguity
 
 # issue #6175
 function g6175(); print(""); (); end

--- a/test/inline.jl
+++ b/test/inline.jl
@@ -60,12 +60,12 @@ test_inlined_symbols(test_outer, Tuple{Int64})
         return y
     end
 end
-function bar()
+function bar12620()
     for i = 1:3
         foo_inl(i==1)
     end
 end
-@test_throws UndefVarError bar()
+@test_throws UndefVarError bar12620()
 
 # issue #16165
 @inline f16165(x) = (x = UInt(x) + 1)

--- a/test/show.jl
+++ b/test/show.jl
@@ -412,8 +412,7 @@ test_mt(show_f1, "show_f1(x...)")
 test_mt(show_f2, "show_f2(x...)")
 test_mt(show_f3, "show_f3(x...)")
 test_mt(show_f4, "show_f4(x::Vararg{Any,3})")
-intstr = string(Int)
-test_mt(show_f5, "show_f5{T,N}(A::AbstractArray{T,N}, indexes::Vararg{$intstr,N})")
+test_mt(show_f5, "show_f5{T,N}(A::AbstractArray{T,N}, indexes::Vararg{$Int,N})")
 
 # Issue #15525, printing of vcat
 @test sprint(show, :([a;])) == ":([a;])"

--- a/test/threads.jl
+++ b/test/threads.jl
@@ -129,7 +129,7 @@ module M14726_2
 using Base.Test
 using Base.Threads
 @threads for i in 1:100
-    # Make sure current module is the same with the one on the thread that
+    # Make sure current module is the same as the one on the thread that
     # pushes the work onto the threads.
     # The @test might not be particularly meaningful currently since the
     # thread infrastructures swallows the error. (See also above)

